utils/src/wizard/AzureWizardPromptStep.ts

```diff
@@ -14,11 +14,17 @@ export abstract class AzureWizardPromptStep<T extends types.IActionContext> impl
     public numSubExecuteSteps: number;
     public propertiesBeforePrompt: string[];
     public prompted: boolean;
+    /**
+     * If step doesn't show any UI in the prompt method,
+     * but still needs to be undone when the user goes back.
+     */
+    public skipped: boolean;
```

Comment:
It seems like the wizard framework shouldn't need to know about whether a step actually showed UI in its prompt. The `SkipIfOne` steps are still "prompted", it's just that nothing shows up in the UI, so they should still be undone normally.

Reply:
> It seems like the wizard framework shouldn't need to know about whether a step actually showed UI in its prompt. The `SkipIfOne` steps are still "prompted", it's just that nothing shows up in the UI, so they should still be undone normally.

The issue is that if a step doesn't show UI, if a user goes back, it will just immediately skip that step and return to the same step the user was just on.

Step 1 (skip if one, no UI shown) > Step 2 (current step, user hits back button)

User hits back button on Step 2, we go back to Step 1, but it just gets skipped and we're back to Step 2.

> It looks like `step.prompted` is being set based on when the actual UI part of the prompt step finishes, which is insane (it assumes exactly one prompt in every step, with no allowance for zero or more than one).

For context, this change was made in https://github.com/microsoft/vscode-azuretools/pull/688. It was to fix some a bug with the back button, see PR for details.

> It should just set `prompted = true` when the `prompt` Promise resolves.

I agree, we should make that change. The `skipped` property I added should take care of this bug. So to avoid regressions on back button behavior, we'll need to set the skipped property for steps that don't show UI.